cluster/gce/gci/configure-helper.sh

```diff
@@ -83,16 +139,71 @@ function safe-format-and-mount() {
   mount -o discard,defaults "${device}" "${mountpoint}"
 }
 
-# Local ssds, if present, are mounted at /mnt/disks/ssdN.
+# Creates a bind mounting from the /mnt/disks/ fs mounted folder to
+# a new mountpoint in /mnt/disks/by-uuid/ using the devices UUID 
+function unique-uuid-bind-mount(){
+  local device=$1
+  local interface=$2
+  local mountpoint=$3
+
+  # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
+  udevadm control --reload-rules
+  udevadm trigger
+  udevadm settle
+
+  # Get the real UUID of the device
+  local item=$(readlink -f ${device} | cut -d '/' -f 3)
+  local myuuid=$(ls -l /dev/disk/by-uuid/ | grep ${item} | tr -s ' ' | cut -d ' ' -f 9) 
+  local bindpoint="${UUID_MNT_PREFIX}-${interface}-fs/local-ssd-${myuuid}"
+
+  # Mount device to the mountpoint
+  mkdir -p "${bindpoint}"
+  echo "Binding '${device}' at '${bindpoint}'"
+  mount --bind "${mountpoint}" "${bindpoint}"
+  chmod a+w ${bindpoint}
+
+}
+
+# Mounts, bindmounts, or symlinks depending on the interface and format
+# of the incoming device
+function mount-ext(){
+  local ssd=$1
+  local interface=$2
+  local format=$3  
+
+  if [[ ${format} == "fs" ]]; then
+    if [[ ${interface} == "scsi" ]]; then
+      local scsinum=`echo ${ssd} | sed -e 's/\/dev\/disk\/by-id\/google-local-ssd-\([0-9]*\)/\1/'`
+      local mountpoint="/mnt/disks/ssd${scsinum}"
+    else
+      local nvmenum=`echo ${ssd} | sed -e 's/\/dev\/disk\/by-id\/google-nvme0n\([0-9]*\)/\1/'`
+      local mountpoint="/mnt/disks/ssd-nvme${nvmenum}"
+    fi
+    safe-format-and-mount ${ssd} ${mountpoint}
+    unique-uuid-bind-mount ${ssd} ${interface} ${mountpoint}
+  elif [[ ${format} == "block" ]]; then
+    safe-block-symlink ${ssd} ${interface}
+  else
+    echo "Disk format must be either fs or block, got ${format}"
+  fi
+}
+
+# Local ssds, if present, are mounted or symlinked to their appropriate
+# locations
 function ensure-local-ssds() {
+  get-local-disk-num "scsi" "block"
+  local scsiblocknum=${total}
+  local i=0
   for ssd in /dev/disk/by-id/google-local-ssd-*; do
```

Comment:
Should we sort this list to ensure that we process the disks in the same order?

Reply:
The ${ssd} will already be sorted by default. I do remember why i decided to do "fs" disks first now though, now the disk numbers will make even less sense because they would start with the numbers after block storage.

Ex. We have 2 block and 2 fs devices
google-local-ssd-[0,1] are block,
then when we mount the fs devices we will have them mounted at:
/mnt/disks/ssd2 /mnt/disks/ssd3

And if we include nvme devices in the mix they might not even be increasing by 1 each time...

Can be solved by re-indexing them to start at 0 and monotonically increment. But then the numbers are no longer 1-1 from /dev/disks to /mnt/disks